framework/src/main/java/org/checkerframework/idesupport/TypeInformationPresenter.java

```diff
@@ -0,0 +1,520 @@
+                            : factory.getAnnotatedType(tree.getVariable());
+            reportTreeType(tree, varType, MessageKind.ASSIGN_LHS_DECLARED_TYPE);
+            reportTreeType(
+                    tree,
+                    factory.getAnnotatedType(tree.getExpression()),
+                    MessageKind.ASSIGN_RHS_TYPE);
+            return super.visitAssignment(tree, unused);
+        }
+
+        @Override
+        public Void visitCompoundAssignment(CompoundAssignmentTree tree, Void unused) {
+            reportTreeType(tree, factory.getAnnotatedType(tree));
+            AnnotatedTypeMirror varType =
+                    genFactory != null
+                            ? genFactory.getAnnotatedTypeLhs(tree.getVariable())
+                            : factory.getAnnotatedType(tree.getVariable());
+            reportTreeType(tree, varType, MessageKind.ASSIGN_LHS_DECLARED_TYPE);
+            reportTreeType(
+                    tree,
+                    factory.getAnnotatedType(tree.getExpression()),
+                    MessageKind.ASSIGN_RHS_TYPE);
+            return super.visitCompoundAssignment(tree, unused);
+        }
+
+        @Override
+        public Void visitUnary(UnaryTree tree, Void unused) {
+            Tree.Kind treeKind = tree.getKind();
+            switch (treeKind) {
+                case UNARY_PLUS:
+                case UNARY_MINUS:
+                case BITWISE_COMPLEMENT:
+                case LOGICAL_COMPLEMENT:
+                case PREFIX_INCREMENT:
+                case PREFIX_DECREMENT:
+                    reportTreeType(tree, factory.getAnnotatedType(tree));
+                    break;
+                case POSTFIX_INCREMENT:
+                case POSTFIX_DECREMENT:
+                    reportTreeType(tree, factory.getAnnotatedType(tree));
+                    if (genFactory != null) {
+                        reportTreeType(
+                                tree,
+                                genFactory.getAnnotatedTypeRhsUnaryAssign(tree),
+                                MessageKind.ASSIGN_RHS_TYPE);
+                    }
+                    break;
+                default:
+                    throw new BugInCF(
+                            "Unsupported unary tree type "
+                                    + treeKind
+                                    + " for "
+                                    + TypeInformationPresenter.class.getCanonicalName());
+            }
+            return super.visitUnary(tree, unused);
+        }
+
+        @Override
+        public Void visitBinary(BinaryTree tree, Void unused) {
+            switch (tree.getKind()) {
+                case LESS_THAN:
+                case GREATER_THAN:
+                case LESS_THAN_EQUAL:
+                case GREATER_THAN_EQUAL:
+                case EQUAL_TO:
+                case NOT_EQUAL_TO:
+                    // don't report if tree is a comparison operator
```

Comment:
I initiall didn't find any reasons to show the types of comparisons, and yes, it could be useful in some minor cases. I have changed the code to display type information for comparisons.